### PR TITLE
Add reporter.Monitor() for goroutines

### PR DESCRIPTION
**Pull request by v-yarotsky**
@bmarini @sanjayprabhu

### CODE REVIEW

File: reporter/reporter_test.go [hunk 1]
```diff
@@ -39,3 +40,33 @@ func TestReport(t *testing.T) {
...
+		defer Monitor(ctx)
+		panic("oh noes!")
+	}()
+	select {
+	case <-done:
+		if reportedError == nil || reportedError.Error() != "panic: oh noes!" {
+			t.Errorf("expected panic 'oh noes!' to be reported, got %#v", reportedError)
+		}
+	case <-time.After(3 * time.Second):
+		t.Errorf("well well")
```
Comment (bmarini) [inline reporter/reporter_test.go]: cute, but more descriptive error message please? :)

**Comment by sanjayprabhu** [comment]
:+1:

File: reporter/reporter_test.go (hunk 1 above)
Comment (v-yarotsky) [inline reporter/reporter_test.go]: oops sorry :)

**Comment by bmarini** [comment]
Looks great :+1:

**Pull request merged**